File: app/src/main/java/im/tox/antox/utils/IconColor.java
package im.tox.antox.utils;

import android.graphics.Color;
import im.tox.jtoxcore.ToxUserStatus;

/**
 * Created by ollie on 30/05/14.
 */
public class IconColor {
    public static String iconColorAsString(Boolean isOnline, ToxUserStatus status)
    {
        String color;
        if (!isOnline) {
            color = "#B0B0B0"; //offline
        } else if (status == ToxUserStatus.TOX_USERSTATUS_NONE) {
            color = "#5ec245"; //online
        } else if (status == ToxUserStatus.TOX_USERSTATUS_AWAY) {
            color = "#E5C885"; //away
        } else if (status == ToxUserStatus.TOX_USERSTATUS_BUSY) {
            color = "#CF4D58"; //busy
        } else {
            color = "#FFFFFF";
        }
        return color;
    }

    public static int iconColorAsColor(Boolean isOnline, ToxUserStatus status)
    {
        return Color.parseColor(iconColorAsString(isOnline,status));

    }
}

File: app/src/main/scala/im/tox/antox/utils/IconColor.scala
package im.tox.antox.utils

import im.tox.antoxnightly.R
import im.tox.tox4j.core.enums.ToxStatus

object IconColor {

  def iconDrawable(isOnline: java.lang.Boolean, status: ToxStatus): Int = {
    val color = if (!isOnline) {
      R.drawable.circle_offline
    } else if (status == ToxStatus.NONE) {
      R.drawable.circle_online
    } else if (status == ToxStatus.AWAY) {
      R.drawable.circle_away
    } else if (status == ToxStatus.BUSY) {
      R.drawable.circle_busy
    } else {
      R.drawable.circle_offline
    }
    color
  }

}
